model/src/test/java/com/ge/predix/acs/rest/AttributeResponseTest.java

```diff
@@ -0,0 +1,43 @@
+package com.ge.predix.acs.rest;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.ge.predix.acs.model.Attribute;
+import com.ge.predix.acs.rest.attribute.adapter.AttributesResponse;
+
+public class AttributeResponseTest {
+
+    @Test
+    public void testCheckSizeLimit() {
+        AttributesResponse attributesResponse = createAttributesResponse(1);
+        Assert.assertEquals(attributesResponse.checkSizeLimit(100, 100), 17);
+    }
+
+    @Test
+    public void testCheckSizeLimitMaxAttributes() {
+        AttributesResponse attributesResponse = createAttributesResponse(2);
+        Assert.assertEquals(attributesResponse.checkSizeLimit(1, 100), Integer.MAX_VALUE);
+    }
+
+    @Test
+    public void testCheckSizeLimitMaxAttributesBytes() {
+        AttributesResponse attributesResponse = createAttributesResponse(1);
+        Assert.assertEquals(attributesResponse.checkSizeLimit(100, 1), Integer.MAX_VALUE);
+
+    }
+
+    private AttributesResponse createAttributesResponse(final int numberOfAttributes) {
+        AttributesResponse attributesResponse = new AttributesResponse();
+        Set<Attribute> attributeSet = new HashSet<>();
+        for (int i = 0; i < numberOfAttributes; i++) {
+            attributeSet.add(new Attribute("issuer" + Integer.toString(i), "name" + Integer.toString(i),
+                    "value" + Integer.toString(i)));
+        }
+        attributesResponse.setAttributes(attributeSet);
+        return attributesResponse;
```

Comment:
`return new AttributesResponse().setAttributes(attributeSet);` should get rid of the local variable?